packages/aws-cdk-lib/aws-codepipeline-actions/test/stepfunctions/stepfunctions-invoke-actions.test.ts

```diff
@@ -2,153 +2,160 @@ import { Template, Match } from '../../../assertions';
-              },
-              Effect: 'Allow',
+    Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
+      PolicyName: 'MyPipelineInvokeCodePipelineActionRoleDefaultPolicy07A602B1',
+      PolicyDocument: {
+        Statement: Match.arrayWith([
+          {
+            Action: ['states:StartExecution', 'states:DescribeStateMachine'],
+            Resource: {
+              Ref: 'SimpleStateMachineE8E2CF40',
             },
-          ]),
-        },
-      });
+            Effect: 'Allow',
+          },
+        ]),
+      },
+    });
 
-      Template.fromStack(stack).resourceCountIs('AWS::IAM::Role', 4);
+    Template.fromStack(stack).resourceCountIs('AWS::IAM::Role', 4);
 
-    });
+  });
 
-    test('Allows the pipeline to describe this stepfunction execution', () => {
-      const stack = new Stack();
+  test('Allows the pipeline to describe this stepfunction execution', () => {
+    const stack = new Stack();
 
-      minimalPipeline(stack);
+    minimalPipeline(stack);
 
-      Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
-        PolicyDocument: {
-          Statement: [
-            {},
-            {
-              Action: 'states:DescribeExecution',
-              Resource: {
-                'Fn::Join': [
-                  '',
-                  [
-                    'arn:',
-                    {
-                      Ref: 'AWS::Partition',
-                    },
-                    ':states:',
-                    {
-                      Ref: 'AWS::Region',
-                    },
-                    ':',
-                    {
-                      Ref: 'AWS::AccountId',
-                    },
-                    ':execution:',
-                    {
-                      'Fn::Select': [
-                        6,
-                        {
-                          'Fn::Split': [
-                            ':',
-                            {
-                              Ref: 'SimpleStateMachineE8E2CF40',
-                            },
-                          ],
-                        },
-                      ],
-                    },
-                    ':*',
-                  ],
+    const arnParts = {
+      'Fn::Split': [':', { Ref: 'SimpleStateMachineE8E2CF40' }],
+    };
+    Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {},
+          {
+            Action: 'states:DescribeExecution',
+            Resource: {
+              'Fn::Join': [
+                '',
+                [
+                  'arn:',
+                  { 'Fn::Select': [1, arnParts] },
+                  ':states:',
+                  { 'Fn::Select': [3, arnParts] },
+                  ':',
+                  { 'Fn::Select': [4, arnParts] },
+                  ':execution:',
+                  { 'Fn::Select': [6, arnParts] },
+                  ':*',
```

Comment:
This assertion has changed due to the change in how the executions ARN is now computed "correctly".